code-challenges.test.js

```diff
@@ -14,48 +14,134 @@
+        var secretCodeWord2 = "Gobbledygook"
+        // Expected output: "G0bbl3dyg00k" 
+        expect(sercretCode(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+        var secretCodeWord3 = "Eccentric"
+        // Expected output: "3cc3ntr1c"
+        expect(sercretCode(secretCodeWord3)).toEqual("3cc3ntr1c")
+    })
+})
 // b) Create the function that makes the test pass.
+const sercretCode = (string) => {
 
+    const secretArr = string.split('');
+    const codedMessage = secretArr.reduce((accum, currVal) => {
 
+        if(currVal == 'a' || currVal == 'A') {
+            accum = accum + '4';
+        }else if (currVal == 'e' || currVal == 'E') {
+            accum = accum + '3';
+        }else if (currVal == 'i' || currVal == 'I') {
+            accum = accum + '1';
+        }else if (currVal == 'o' || currVal == 'O') {
+            accum = accum + '0';
+        }else {
+            accum = accum + currVal;
+        }
 
-// --------------------2) Create a function that takes in an array and returns all the words that contain the letter a.
+        return accum;
 
-// a) Create a test with an expect statement using the variable provided.
+    }, '');
+    
+
+    return codedMessage;
 
-var arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
-// Expected output: ["Apple", "Banana", "Orange"]
-var arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
-// Expected output: ["Mango", "Apricot", "Peach"]
+}
 
 
 
+// --------------------2) Create a function that takes in an array and returns all the words that contain the letter a.
+/////////////////////////////////////////////////// Todd's Pseudo Code ///////////////////////////////////////////////////
+// You guys of always taught us to look at the question and break it down
+// So the first step is to create a function findingLetter, Parmeter - array, Iterate - .forEach.
+// Now that I have my functionin it was able to go through the array, find all the words 
+// that have the letter a in them and display them.
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// a) Create a test with an expect statement using the variable provided.
+describe("findingLetter", () => {
+    test("takes in an array and returns all the words that contain the letter a.", () => {
+       var arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
+       // Expected output: ["Apple", "Banana", "Orange"]
+       expect(findingLetter(arrayOfWords1)).toEqual(["Apple", "Banana", "Orange"])
+       var arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
+       // Expected output: ["Mango", "Apricot", "Peach"]
+       expect(findingLetter(arrayOfWords2)).toEqual(["Mango", "Apricot", "Peach"])
+    })
+})
 // b) Create the function that makes the test pass.
+const findingLetter = (arr) => {
+    let containsLetterA = [];
+    arr.forEach(word => {
```

Comment:
This is an effective use for .forEach() but JS wants to be nicer to you than that and gives us .filter() which works really well in this situation. Great work though figuring this out!